test/SILOptimizer/diagnostic_constant_propagation.swift

```diff
@@ -1,342 +1,340 @@
+    // -1 * 2^1024 * (2^54-1)/2^54
+    var _/*float64_min_first_overflow*/      : Float64 = (-179769313486231580793728971405303415079934132710037826936173778980444968292764750946649017977587207096330286416692887910946555547851940402630657488671505820681908902000708383676273854845817711531764475730270069855571366959622842914819860834936475292719074168444365510704342711559699508093042880177904174497792) // expected-error {{integer literal '-179769313486231580793728971405303415079934132710037826936173778980444968292764750946649017977587207096330286416692887910946555547851940402630657488671505820681908902000708383676273854845817711531764475730270069855571366959622842914819860834936475292719074168444365510704342711559699508093042880177904174497792' overflows when stored into 'Double'}}
+    
+    // -1 * 2^1024
+    var _/*float64_min_definitely_overflow*/ : Float64 = (-179769313486231590772930519078902473361797697894230657273430081157732675805500963132708477322407536021120113879871393357658789768814416622492847430639474124377767893424865485276302219601246094119453082952085005768838150682342462881473913110540827237163350510684586298239947245938479716304835356329624224137216) // expected-error {{integer literal '-179769313486231590772930519078902473361797697894230657273430081157732675805500963132708477322407536021120113879871393357658789768814416622492847430639474124377767893424865485276302219601246094119453082952085005768838150682342462881473913110540827237163350510684586298239947245938479716304835356329624224137216' overflows when stored into 'Double'}}
 }
 
 @_transparent 
 func intConversionWrapperForUSCheckedConversion(_ x: UInt8, _ unused: UInt8) -> Int8 {
-  return Int8(x)
+    return Int8(x)
 }
 @_transparent 
 func intConversionWrapperForLiteral() -> Int8 {
-  return 255 // expected-error {{integer literal '255' overflows when stored into 'Int8'}}
+    return 255 // expected-error {{integer literal '255' overflows when stored into 'Int8'}}
 }
 func testFallBackDiagnosticMessages() {
-  _ = intConversionWrapperForUSCheckedConversion(255, 30) // expected-error {{integer overflows when converted from unsigned 'Builtin.Int8' to signed 'Builtin.Int8'}}
-  _ = intConversionWrapperForLiteral() // expected-error {{integer literal '255' overflows when stored into signed 'Builtin.Int8'}}
+    _ = intConversionWrapperForUSCheckedConversion(255, 30) // FIXME: false negative: uncaught overflow error. See the function `intConversoin..` definition (see also <rdar://problem/39120081>).
+    _ = intConversionWrapperForLiteral() // expected-error {{integer literal '255' overflows when stored into signed 'Builtin.Int8'}}
 }
 
-// XXX FIXME -- blocked by: 15735295 Need [su]{div,rem}_with_overflow IR
-/*
+// TODO: There is a radar related to these: <rdar://problem/15735295>. But, everything seem to work as expected. Is the problem fixed?
```

Comment:
Yes, it looks like it is. You can remove the TODO. Thanks for re-enabling these!